### PR TITLE
Prefer string.Empty to ""

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/NotifyIcon.cs [hunk 1]
```diff
@@ -47,15 +47,15 @@ public sealed class NotifyIcon : Component {
         private object syncObj = new object();
 
         private Icon icon = null;
-        private string text = "";
+        private string text = string.Empty;
         private int id = 0;
         private bool added = false;
```
Comment (danmosemsft) [inline src/System.Windows.Forms/src/System/Windows/Forms/NotifyIcon.cs]: We should have an issue to remove all these redundant field initializations. Oddly, Roslyn doesn't remove them.

**Comment by stephentoub** [comment]
> at JIT time the difference is immeasurable

That's because the JIT generates the same assembly for both.

> I have always assumed they are all interned to a single instance

They are.

> do we have a policy for using string.Empty over ""

Not really.

**Comment by danmosemsft** [comment]
> That's because the JIT generates the same assembly for both.

Right, I got that, by "immeasurable" I meant, the difference between JITing a field load and a constant load.

OK @zsd4yr up to you - do you want to take this change or not? We do not have a policy in the CoreFX style guide. I see use of both. I am actually not sure why `string.Empty` exists - perhaps from early days when `""` was not reliably interned.

**Comment by stephentoub** [comment]
> I meant, the difference between JITing the IL for a field load and a constant load.

Maybe we're saying the same thing, I'm not sure.  But the assembly for both is literally identical:
https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAGABJgRgG4BYAKEwGZ8AmXAYVwG9zc39rC8BZAgCgCUuALwA+XAGU4AJwCWAOwDmAOgCiqAA5wAnqTLsO+AjxqCR4gEQW9AXyA=
```C#
using System;
public class C {
    public string M1() => String.Empty;
    public string M2() => "";
}
```
results in
```
C..ctor()
    L0000: ret

C.M1()
    L0000: mov rax, 0x19fa02b3620
    L000a: mov rax, [rax]
    L000d: ret

C.M2()
    L0000: mov rax, 0x19fa02b3620
    L000a: mov rax, [rax]
    L000d: ret
```

**Comment by RussKie** [comment]
I think I've read it at either @jaredpar's or Eric Lippert's blogs that there was a difference between the two, and whilst largely it is the same thing in some use cases it may not be.

**Comment by sharwell** [comment]
If this is going to be a rule, you might as well enable [SA1122](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/57821a1c4d014e8d80b3e236ff444e0e30d8a112/documentation/SA1122.md) so it's automated.

https://github.com/dotnet/winforms/blob/ee3a37b37de437d04f542e650bd5c26e504b73cc/eng/Shipping.ruleset#L62

**Comment by jaredpar** [comment]
@RussKie 

> I think I've read it at either @jaredpar's or Eric Lippert's blogs that there was a difference between the two, and whilst largely it is the same thing in some use cases it may not be.

I believe this was true way, way back in the 2006 days. Something to do with interning or the like. These days though there isn't an observable different I'm aware of.

**Comment by RussKie** [comment]
Thank you 👍

**Comment by danmosemsft** [comment]
@stephentoub yes we were saying the same thing.
@zsd4yr this is over to you to pick what you want in the repo, and possibly enable the stylecop rule too.

File: src/System.Windows.Forms/src/System/Windows/Forms/NotifyIcon.cs (hunk 1 above)
Comment (weltkante) [reply src/System.Windows.Forms/src/System/Windows/Forms/NotifyIcon.cs]: #925

**Comment by hughbe** [comment]
I did this PR because I noticed that in PR feedback I was getting comments saying “use string.empty” so just went and fixed this across Winforms!

**Comment by danmosemsft** [comment]
Ping @zsd4yr for question above.

**Comment by hughbe** [comment]
> /eng/ci.yml: Could not find /eng/common/templates/jobs/jobs.yml in repository self hosted on https://api.github.com using commit 48fbc9d2bc9f037dc54fa55263013ea94142bc45. GitHub reported the error, "Not Found"

**Comment by zsd4yr** [comment]
Hey all.

My preference is for string.empty; perhaps it would be best to make a huge pass through the codebase?

I would prefer not to write a stylecop rule; is there any way to do this in .editorconfig or gitattributes? hmmm...

**Comment by sharwell** [comment]
> I would prefer not to write a stylecop rule unless it can be automatically fixed by stylecop as well;

The rule already exists. You just have to turn it on by changing the line I referenced from `None` to `Warning`. It contains an extremely fast and accurate Fix All feature.

**Comment by hughbe** [comment]
> My preference is for string.empty; perhaps it would be best to make a huge pass through the codebase?

Thats what I did :)

**Comment by zsd4yr** [comment]
> > I would prefer not to write a stylecop rule unless it can be automatically fixed by stylecop as well;
> 
> The rule already exists and is already imported by the winforms build. You just have to turn it on by changing the line I referenced from `None` to `Warning` (**or preferably** just deleting the line and letting it use the default). It contains a fast and accurate Fix All feature.

@hughbe would you mind doing that and "fixing all"? Thanks @sharwell, I did not know that

**Comment by AdamYoblick** [comment]
The reason I always learned for preferring string.Empty over the literal "" string is literal strings are always more error-prone. 

At a quick glance, would you really see "'" has a single quote in the middle of it? Maybe, maybe not. But that could be a runtime error down the line. Whereas string.Empty is very clear, and you can't typo it or the compiler will get you. 

Not to mention that it also meshes nicely with many of the structs that can't be null that often have an Empty static property used for initialization.

**Comment by jaredpar** [comment]
Remember though, `string.Empty` is clear while `String.Empty` is not. 😄

**Pull request merged**